e2e/test/scenarios/models/reproductions.cy.spec.ts

```diff
@@ -289,3 +292,67 @@ describe("issue 23103", () => {
     hovercard().findByText("4 distinct values").should("exist");
   });
 });
+
+describe("issue 39150", () => {
+  const ccName = "CC Rating";
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it(
+    "shows allows custom columns with the same name in nested models (metabase#39150)",
+    { viewportWidth: 1600 },
+    () => {
+      createQuestion({
```

Comment:
I think there are 2 slightly different ways to reproduce the issue and they both deserve a test:
- custom column being named like an aggregation column in source question (original repro in [#39150](https://github.com/metabase/metabase/issues/39150))
- custom column being named like a custom column in source question ([comment](https://github.com/metabase/metabase/issues/39150#issuecomment-2029796451))

Reply:
The issue was ambiguous to me with poor repro steps, and I didn't interpret the first case as the "custom column being named like an aggregation column in source question" but I can add that for sure.